### DISCUSSION TITLE
Font customization for tournament view

### DISCUSSION THREAD

**Discussion by VINXIS** [ppy/osu#13206] [Ideas]
**Describe the new feature:** Be able to select our own font choices that are installed in the PC already instead of being limited to the font(s) used by the client currently. This is in order to provide more design customizability for tournaments to have the tournament view be able to fit more into the aesthetic choices that were made for the tournaments that are using this.

**Proposal designs of the feature:** Could add a dropdown here that allows the selection of being able to change the font so the tournament streamer can just toggle their font choice there and go ahead with the rest of their customization
![](https://i.imgur.com/f4m8iHP.png)

**Comment by peppy** [comment]
Fonts from PC will likely never be available. You would need to first convert then to a skin format manually.

**Reply by sr229** [reply]
I would love to know what's the technical constraints behind this and if it's possible to do on-the-fly rasterization for a selected font?

**Reply by smoogipoo** [reply]
Of course, there's many ways to rasterize fonts. Best implemented as a separate module from o!f to start off with though.

**Reply by peppy** [reply]
No technical constraint. Priorities, UX and licensing considerations.

**Comment by LeNitrous** [comment]
At the very least, if font color is able to be changed. o!m 4k FL4T as their designs heavily rely on light colors causing this to happen:
![image](https://user-images.githubusercontent.com/20495991/119759965-c3619680-bedb-11eb-9f05-a0cf2eb0a8fc.png)

And probably for long term, which is probably outside the scope of this issue, make the layout be customizable since skin editing (or at the very least moving around elements) is possible now.

**Comment by peppy** [comment]
@LeNitrous you are a developer. Please consider building a build of lazer for your tournament with a single word change to change the colour.

The long path forward is to first establish the skin layout editor in the game to have these features. It would then be used in the tournament edit screens in the same way.

**Comment by VINXIS** [comment]
Skin usage also sounds like itd work well too
